cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -286,6 +289,53 @@ file(
     "${using_mpi}\n"
 )
 
+# handle requested python version if such is specified
+hunter_parse_cmake_args_for_keyword(
+  CMAKE_ARGS "@HUNTER_Boost_CMAKE_ARGS@"
+  KEYWORD REQUESTED_PYTHON_VERSION
+  OUTPUT requested_python
+)
+
+# set up paths for python in boost.user.jam
+if(NOT requested_python STREQUAL "")
+  find_package(PythonInterp ${requested_python} EXACT QUIET)
+  if(PYTHONINTERP_FOUND)
+    set(python_executable ${PYTHON_EXECUTABLE})
+    execute_process(
+      COMMAND "${python_executable}" -c "import os.path, sysconfig; info = sysconfig.get_paths(); print(info['include'])"
+      RESULT_VARIABLE python_process
+      OUTPUT_VARIABLE python_include_directory
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(NOT python_process EQUAL 0)
+      unset(python_include_directory)
+    endif()
+  endif()
+  if(python_executable)
+    execute_process(
+      COMMAND "${python_executable}" -c "import os.path, sysconfig; info = sysconfig.get_paths(); print(os.path.dirname(info['stdlib']))"
+      RESULT_VARIABLE python_process
+      OUTPUT_VARIABLE python_library_directory
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(NOT python_process EQUAL 0)
+      unset(python_library_directory)
+    endif()
+  endif()
+
+  if(python_executable AND python_include_directory AND python_library_directory)
+    file(
+        APPEND ${boost_user_jam}
+        "using python  : ${requested_python} : \"${python_executable}\" : \"${python_include_directory}\" : \"${python_library_directory}\" ;\n"
+    )
+  else()
+    hunter_user_error("Could not find some or all required components for the requested Python version ${requested_python}.")
```

Comment:
There seems to be 3 different errors. Both users experience and code will be better if you will check them separately:
```cmake
if(NOT PYTHONINTERP_FOUND)
  hunter_user_error("Python not found")
endif()
````

```cmake
if(NOT python_process EQUAL 0)
  hunter_user_error("include dir not found")
endif()
```

```cmake
if(NOT python_process EQUAL 0)
  hunter_user_error("library dir not found")
endif()
```